### ISSUE TITLE
[$2000] Android/IOS - Split/request money - Chat composer is focused after IOU request from green plus button

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#10731]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to App and login with any account
2. Click the green plus button and click Request money
Enter an amount and select the `Next` confirmation button
Complete the request money flow

## Expected Result:
Always focus the input, but prevent keyboard from opening on mobile & mWeb

## Actual Result:
The composer is focused.

## Workaround:
Swipe the keyboard down to un-focus the composer

## Platform:

Where is this issue occurring?

- iOS
- Android

**Version Number:** 1.1.95.4

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):** any

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/187747715-3711327d-e3d4-4198-9df5-32400f8cb46e.mp4

https://user-images.githubusercontent.com/93399543/187747751-f8f2a162-cb62-4b85-91cc-d9bf024a13ba.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by neil-marcellini** [comment]
Hmm, so if you navigate to any normal chat on mobile, the composer is not focused until you tap into it. That is the desired behavior. I think this is a valid issue if we change it from:
"Chat composer is not focused after IOU request" to "Chat composer is focused after IOU request from green plus button"

Also it looks like this only happens when the request is with a brand new chat. I was not able to see the issue when sending a request via the green plus to a chat where there was a good amount of history and zero IOU requests. This is a good external issue with lower priority (weekly).

**Comment by neil-marcellini** [comment]
I updated the issue title and description to reflect the actual problem.

**Comment by bfitzexpensify** [comment]
Upwork job [here](https://www.upwork.com/jobs/~0171651b178a420568)

**Comment by syedsaroshfarrukhdot** [comment]
**Proposal**

https://github.com/Expensify/App/blob/8e07c62fb9780c627106f45eb72614466a850f27/src/pages/home/report/ReportActionCompose.js#L595

This issue is caused by _.size(this.props.reportActions) === 1 in autofocus props of Composer when this.props.reportActions === 1 is true it focuses in the input field to fix the issue we need to remove it.

**Solution :-**
To fix the issue we need to set autofocus props like this autoFocus={this.shouldFocusInputOnScreenFocus}

**After Fixing :-**

https://user-images.githubusercontent.com/81307212/187970668-d2c6ef18-7ab4-41e9-8e42-a48253a8ec7f.mov

**Comment by christianwen** [comment]
# **Proposal**

## **Problem**
The issue is in `src/pages/home/report/ReportActionCompose.js`.

This is due to when we split/request money for the first time for a new group/person, the `this.props.reportActions` is loaded with only 1 `IOU` message initially and then it will be populated with other messages like the `CREATED` message for the group. Thus, initially the `_.size(this.props.reportActions) === 1` will be true (before it loads everything) and the composer will be focused incorrectly.

The previous proposal to remove the `_.size(this.props.reportActions) === 1` will cause a regression because that code is there so that when user enters a new chat, the keyboard will be focused automatically. We can refer to the requirement for that here https://github.com/Expensify/App/pull/4921#discussion_r718023887.

## **Solution**
We'll implement a new `isNewChat` method that checks both `_.size(this.props.reportActions) === 1` and the only message in the chat is the `CREATED` message, this will help us avoid the lazy loading problem mentioned.
```diff
+isNewChat() {
+        if (_.size(this.props.reportActions) === 1) {
+            const firstReportAction = _.head(_.values(this.props.reportActions));
+            return firstReportAction.actionName === CONST.REPORT.ACTIONS.TYPE.CREATED;
+        }

+        return false;
+    }
....
<Composer
-      autoFocus={this.shouldFocusInputOnScreenFocus || _.size(this.props.reportActions) === 1}
+      autoFocus={this.shouldFocusInputOnScreenFocus || this.isNewChat()}
```

After the fix, the chat composer is no longer focused incorrectly

https://user-images.githubusercontent.com/21312517/188118275-8c39c491-5048-4b85-8e31-240e0f5b3666.mp4

I also discover another bug where the Chat composer will initially have the `Say hello!` placeholder (which should only appear if the chat is completely new and there's no message in the chat), only a few moments later it will change to `Write something...` placeholder. You can see the issue in the video capture attached in the issue description:

<img width="290" alt="Screen Shot 2022-09-02 at 17 17 51" src="https://user-images.githubusercontent.com/21312517/188118610-b5b86a66-9f9e-4489-b98f-088a105f5779.png">

The fix for that will use the same `isNewChat` method I mentioned above, please confirm if I should fix it in the scope of this issue as well.

Please let me know if you have any concerns regarding the above proposal.
Thanks!

**Comment by syedsaroshfarrukhdot** [comment]
As mentioned by @christianwen _.size(this.props.reportActions) === 1 will cause a regression because that code is there so that when a user enters a new chat, the keyboard will be focused automatically. 

But I tested it thoroughly when was proposing a solution the composer was not focused when a user enters a new chat

https://user-images.githubusercontent.com/81307212/188401528-b592ec1b-18c6-47e4-b1d9-b64099f6bb04.mp4

So I thought there is no need of _.size(this.props.reportActions) === 1 so we will be able to resolve the issue by removing this condition. If we need to focus when a user enters a new chat then we could change condition to _.size(this.props.reportActions) === 0 or we need to find a solution to check why _.size(this.props.reportActions) === 1 is not getting correct size of messages on intial render. 

### Solution after _.size(this.props.reportActions) === 0 :-

https://user-images.githubusercontent.com/81307212/188403471-778359ad-bd65-40bd-903b-075297ccaf77.mov

We need to identify first what we want to achieve does the composer needs to be focused on a new chat or not ? After identifying what needs to be done then we can work on a solution. @neil-marcellini

**Comment by neil-marcellini** [comment]
@syedsaroshfarrukhdot thanks for investigating. I think we want to have the composer focused when creating a new chat but not after creating an IOU request. I'll let @iwiznia handle this going forward since he's the CME on this issue.

**Comment by christianwen** [comment]
would it be more suitable for the `composer focused when creating a new chat` to be a separate bug issue since that's not really included in the scope of this issue, or if it's included, we can update the description of this issue to reflect both

**Comment by bfitzexpensify** [comment]
@Santhosh-Sellavel some proposals for you to review above

**Comment by Santhosh-Sellavel** [comment]
@bfitzexpensify Thanks for the bump, But I'm following the discussions and proposals.

Will post my comments by tomorrow.

**Comment by syedsaroshfarrukhdot** [comment]
After testing `this.props.reportActions` i figured out it is nothing to do with `this.props.reportActions` it return always correct value the issue lies with autofocus props of the composer which if `true`, focuses the input on `componentDidMount` and didn't update the focus on `componentDidUpdate` as it is the default behavior of the autofocus props as it takes value always `componentDidMount`. So, I think as it is going to be when ReportScreen Is loaded with no chat `_.size(this.props.reportActions)` is always `0` and after few seconds it updates to `1 `when first message in it the list renders. And when IOURequest is made first time in the new chat it has a  `_.size(this.props.reportActions)` is always `1`.

So to resolve both issue the viable solution i would suggest would be to set 

`_.size(this.props.reportActions) === 0}` is autofocus prop 

https://github.com/Expensify/App/blob/8e07c62fb9780c627106f45eb72614466a850f27/src/pages/home/report/ReportActionCompose.js#L595

After fixing it resolves both issues

https://user-images.githubusercontent.com/81307212/188406145-0d73f2de-af2d-442a-9921-533abc13a947.mp4

**Comment by Santhosh-Sellavel** [comment]
Sorry might need some more time to dig this, will update later today or tomorrow!

**Comment by christianwen** [comment]
I believe the solution to use `_.size(this.props.reportActions) === 0` that relies on the observed behavior of 
> when ReportScreen Is loaded with no chat _.size(this.props.reportActions) is always 0 and after few seconds it updates to 1 when first message in it the list renders

is flaky and not very future proof. Imagine later when it's updated so that the `ReportScreen` is loaded with correct messages list from Onyx initially (instead of having 0 then 1 message after a few seconds), that solution will break.

# **Updated Proposal**
Original proposal: https://github.com/Expensify/App/issues/10731#issuecomment-1235329280

## **Solution**
We'll implement a new `isNewChat` method that checks both `_.size(this.props.reportActions) === 1` and the only message in the chat is the `CREATED` message, this will help us avoid the lazy loading problem mentioned. And the logic to autofocus will be put in the `componentDidUpdate` instead of relying on the `autoFocus` property of the `Composer` which is flaky and does not work as expected when there's rerender.

This will fix both issues:
1. The chat composer will no longer be focused incorrectly after IOU request from green plus button
2. When a new group is created, the chat composer will be focused correctly

```diff
this.state = {
      isFocused: this.shouldFocusInputOnScreenFocus,
      isFullComposerAvailable: props.isComposerFullSize,
      textInputShouldClear: false,
      isCommentEmpty: props.comment.length === 0,
      isMenuVisible: false,
      selection: {
          start: props.comment.length,
          end: props.comment.length,
      },
      maxLines: props.isSmallScreenWidth ? CONST.COMPOSER.MAX_LINES_SMALL_SCREEN : CONST.COMPOSER.MAX_LINES,
+   didAutoFocusOnNewChat: false,
  };
  
 componentDidUpdate(prevProps) {
      ...
+      if (!this.state.didAutoFocusOnNewChat && this.isNewChat()) {
+            this.setDidAutoFocusOnNewChat(true);
+            this.focus();
+        }
      ...
 }
        
+setDidAutoFocusOnNewChat(didAutoFocusOnNewChat) {
+      this.setState({didAutoFocusOnNewChat});
+}

+isNewChat() {
+        if (_.size(this.props.reportActions) === 1) {
+            const firstReportAction = _.head(_.values(this.props.reportActions));
+            return firstReportAction.actionName === CONST.REPORT.ACTIONS.TYPE.CREATED;
+        }

+        return false;
+    }
....
<Composer
-      autoFocus={this.shouldFocusInputOnScreenFocus || _.size(this.props.reportActions) === 1}
+      autoFocus={this.shouldFocusInputOnScreenFocus}
```

1. The chat composer will no longer be focused incorrectly after IOU request from green plus button
https://user-images.githubusercontent.com/21312517/188118275-8c39c491-5048-4b85-8e31-240e0f5b3666.mp4

2. When a new group is created, the chat composer will be focused correctly
https://user-images.githubusercontent.com/21312517/189062189-5b447c22-e3ad-4fa4-93bb-a7782b873581.MP4

**Comment by Santhosh-Sellavel** [comment]
@Expensify/design @iwiznia 

Need some clarifications

What's the expected behavior here?
When should we set focus on the chat composer?
- [ ] On a new chat is created and opened
- [ ] On a new group chat is created and opened
- [ ] On a new chat is created by an IOU request and opened the chat
- [ ] On a new group is created by an IOU Split and opened the chat

Should the focus be set only when the chat is opened for the first time?

Thanks.

**Comment by shawnborton** [comment]
Hmm maybe just when a new chat or group is created and opened, but not after a new IOU or Split is created?

**Comment by shawnborton** [comment]
This might be a good one to bring to the open source room for more visibility though!

**Comment by iwiznia** [comment]
I think it's all of them. If I am left in a chat, I expect to be able to write directly in the chat, if it is not focused on the composer, where is it focused on and is it useful to be focused on that element?

But yeah, maybe bringing the conversation to the slack channel is a good idea.

**Comment by Santhosh-Sellavel** [comment]
Thread is [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1662989446760309) for discussion

**Comment by bfitzexpensify** [comment]
Thanks for raising the discussion @Santhosh-Sellavel. Looks like we're going with this:

>Always focus the input, but prevent keyboard from opening on mobile & mWeb

**Comment by bfitzexpensify** [comment]
Doubled to $500. Reassigning as I head ooo for 2 weeks

**Comment by iwiznia** [comment]
Not overdue, Melvin

**Comment by christianwen** [comment]
# **Proposal**

## **Solution**
We'll make use of `showSoftInputOnFocus` of React Native Text Input and `inputMode` of web input to achieve the intended behavior. 

In `src/pages/home/report/ReportActionCompose.js`
```diff
this.state = {
            ...
+            showSoftInputOnFocus: false,
        };

+setShowSoftInputOnFocus(showSoftInputOnFocus) {
+        this.setState({showSoftInputOnFocus});
+    }

+blur() {
+        if (!this.textInput) {
+              return;
+       }

+       this.textInput.blur();
+}

....
<Composer
// since it will always autofocus
-      autoFocus={this.shouldFocusInputOnScreenFocus || _.size(this.props.reportActions) === 1}
+      autoFocus

+        onTouchStart={() => {
+            if (!this.state.showSoftInputOnFocus) {
+                this.setShowSoftInputOnFocus(true);
                  // after enable keyboard on focus, we need to retrigger focus again for the keyboard to apply
+                this.blur();
+                this.focus();
            }
        }}
```

In `src/components/Composer/index.ios.js`, `src/components/Composer/index.android.js`, `src/components/Composer/index.js`
```diff
const propTypes = {
+      showSoftInputOnFocus: PropTypes.bool,
}

const defaultProps = {
+      showSoftInputOnFocus: true,
}


<RNTextInput
+       showSoftInputOnFocus={this.props.showSoftInputOnFocus}
 />
```

Since `react-native-web` does not support `showSoftInputOnFocus` yet, we need to make an update to support it
In `https://github.com/Expensify/react-native-web` -> `packages/react-native-web/src/exports/TextInput/index.js`

```diff
const {
+    showSoftInputOnFocus
  } = props;
...
  switch (keyboardType) {
    case 'email-address':
      type = 'email';
      break;
    case 'number-pad':
    case 'numeric':
      inputMode = 'numeric';
      break;
    case 'decimal-pad':
      inputMode = 'decimal';
      break;
    case 'phone-pad':
      type = 'tel';
      break;
    case 'search':
    case 'web-search':
      type = 'search';
      break;
    case 'url':
      type = 'url';
      break;
    default:
      type = 'text';
  }

+  if (!showSoftInputOnFocus) {
+    inputMode = 'none';
+  }
```

After the fix, we have the intended behavior, I demo in iOS and mWeb below

iOS
https://user-images.githubusercontent.com/21312517/191922173-bb289d97-c967-49ff-b8f3-4a33b5e2ccfd.mp4

mWeb

https://user-images.githubusercontent.com/21312517/191925165-2466ac2b-fa51-43e8-9fa0-be04c36db82c.MP4


Please let me know if you have any concerns regarding the above proposal.
Thanks!

**Comment by mallenexpensify** [comment]
@Santhosh-Sellavel can you please review @christianwen 's proposal above? Bumping back to `Daily` til then

**Comment by Santhosh-Sellavel** [comment]
Reviewing this now!

**Comment by syedsaroshfarrukhdot** [comment]
@Santhosh-Sellavel 
> Always focus the input, but prevent keyboard from opening on mobile & mWeb

To achieve this we need to introduce a new state such as  inputFocus: false,
our this.state will become

###  this.state = {
            isFocused: this.shouldFocusInputOnScreenFocus,
            isFullComposerAvailable: props.isComposerFullSize,
            textInputShouldClear: false,
            isCommentEmpty: props.comment.length === 0,
            isMenuVisible: false,
            selection: {
                start: props.comment.length,
                end: props.comment.length,
            },
            maxLines: props.isSmallScreenWidth ? CONST.COMPOSER.MAX_LINES_SMALL_SCREEN : CONST.COMPOSER.MAX_LINES,
            inputFocus: false,
        };

We need to write function to clear the inital focus so we can set focus again on basis of our state

 ###  blur() {
         if (!this.textInput) {
          return;
         }
        
        this.textInput.blur();
    }

We need to write a state setter function  such as 

###   setInputFocus(inputFocus) {
                this.setState({inputFocus});
 }

And in Composer we need to pass a props show showSoftInputOnFocus When false, it will prevent the soft keyboard from showing when the field is focused. then we can toggle it onTouchStart. autoFocus will be always true now.
### <Composer
         autoFocus
         showSoftInputOnFocus={this.state.inputFocus}
         onTouchStart={() => {
              if (!this.state.inputFocus) {
                    this.setInputFocus(true);
                    this.blur();
                    this.focus();
                 }
              }}
          multiline


After Fixing :-

https://user-images.githubusercontent.com/81307212/193192692-f443de76-cf6c-4c06-950e-e9cec45f5db0.mov

**Comment by christianwen** [comment]
@Santhosh-Sellavel can you help to add this line? `showSoftInputOnFocus={this.state.showSoftInputOnFocus}`, in here, 
<img width="505" alt="Screen Shot 2022-09-30 at 12 56 29" src="https://user-images.githubusercontent.com/21312517/193201024-defada4a-c083-4ccf-8b06-b16fe209082b.png">

it should work fine

the proposal mentioned in https://github.com/Expensify/App/issues/10731#issuecomment-1263101744 will not work on mWeb because the `showSoftInputOnFocus` is not natively supported in `react-native-web`

**Comment by Santhosh-Sellavel** [comment]
I agree with you @christianwen that's easy to miss!

**Comment by Santhosh-Sellavel** [comment]
@christianwen I verified this seems to work but I noticed some lag in setting focus.

Can you share the screen recording from mWeb, Android, and iOS?

**Comment by mallenexpensify** [comment]
@puneetlath auto-assigned to you, I'm going to be OOO for a month. Thx

**Comment by christianwen** [comment]
@Santhosh-Sellavel the lagging is due to the `onFocus` is triggered on the `Composer` only after the screen finished mounting, so the style of the border is updated later. I believe this is already an existing behavior and not due to the above changes in this issue, in order to fix this we can further modify the following in https://github.com/Expensify/App/blob/9ff60c3a220c9d05001007b2c0dfe0d954623edb/src/pages/home/report/ReportActionCompose.js#L135

```diff
this.state = {
-      isFocused: this.shouldFocusInputOnScreenFocus,
+      isFocused: true,
```
This is ok since the component is always focused when enter the screen for all platforms so we can make `true` the default state to show the correct initial border highlight.

Screen recording after applying the above change:
iOS
https://user-images.githubusercontent.com/21312517/193393980-122451a0-a1da-456e-92fe-4906da879905.mp4

mWeb
https://user-images.githubusercontent.com/21312517/193395820-d02aad41-7192-46dd-a431-91973e56dc4d.mp4

Android
[Android-chat-compose.webm](https://user-images.githubusercontent.com/21312517/193396299-a61d38ba-6314-47ac-ace4-987a00c23812.webm)

**Comment by Santhosh-Sellavel** [comment]
Not overdue

**Comment by mallenexpensify** [comment]
@puneetlath  assigned myself and unassigned you cuz I'm NOT going to be OOO the next month ¯\_(ツ)_/¯

Doubled job price to $1000 https://www.upwork.com/jobs/~0171651b178a420568

**Comment by Santhosh-Sellavel** [comment]
@christianwen proposal mostly looks good. Also need to check this again!

**Comment by iwiznia** [comment]
> Also need to check this again!

@Santhosh-Sellavel so should I review the proposal or not? Not sure what you meant by `check this again`...

**Comment by Santhosh-Sellavel** [comment]
@iwiznia didn't get a chance to test the proposal after this https://github.com/Expensify/App/issues/10731#issuecomment-1264264720 that's why commented to check this again. 

@christianwen 
showSoftInputOnFocus if set `true` keyboard should be shown, not why this is required?

```js
// after enable keyboard on focus, we need to retrigger focus again for the keyboard to apply       
this.blur();
this.focus();
```

**Comment by christianwen** [comment]
@Santhosh-Sellavel `if showSoftInputOnFocus set true keyboard should be shown right` -> this is correct if the input is currently not in focus and then it's focused, the keyboard will be shown.
But in this case, since the input is already focused (it's focused by default when entering the chat screen), when we change `showSoftInputOnFocus` to true it will not pop the keyboard up but will require refocusing in order for the keyboard to open.
You can think of the `showSoftInputOnFocus` as `show soft input on/when focus`, meaning this property will only take effect `onFocus` and not when the TextInput is already focused before.

**Comment by mallenexpensify** [comment]
@Santhosh-Sellavel , can you review @christianwen's comment above plz?

**Comment by Santhosh-Sellavel** [comment]
> if showSoftInputOnFocus is set to true keyboard should be shown right -> this is correct if the input is currently not in focus and then it's focused, the keyboard will be shown.

I disagree as per this doc [here](https://reactnative.dev/docs/textinput#showsoftinputonfocus).
> When false, it will prevent the soft keyboard from showing when the field is focused. The default value is true.

I tried it out without the below lines.
```js
this.blur();
this.focus();
```

It works fine on android, but it doesn't work on iOS.

@christianwen

**Comment by christianwen** [comment]
@Santhosh-Sellavel thanks for highlighting that. The behavior I mentioned is only for iOS, for Android and web it seems to align with your opinion.
Here's the fix for iOS (this should be applied in Expensify's React Native fork, here I show the change in node_modules so it's easy to test locally)
in `node_modules/react-native/Libraries/Text/TextInput/RCTBaseTextInputView.m`:
```diff
(void)setShowSoftInputOnFocus:(BOOL)showSoftInputOnFocus
{
  (void)_showSoftInputOnFocus;
  if (showSoftInputOnFocus) {
    // Resets to default keyboard.
    self.backedTextInputView.inputView = nil;
    
+    // Without the call to reloadInputViews, the keyboard will not change until the textInput field (the first responder) loses and regains focus.
+   if (self.backedTextInputView.isFirstResponder) {
+      [self.backedTextInputView reloadInputViews];
+   }
  } else {
    // Hides keyboard, but keeps blinking cursor.
    self.backedTextInputView.inputView = [UIView new];
  }
}
```

with the above change we can remove the below lines and it still works
```
this.blur();
this.focus();
```

@Santhosh-Sellavel pls try out the above fix and let me know if you have any other concerns
thanks!

**Comment by mallenexpensify** [comment]
Price Doubled to $2000

**Comment by Santhosh-Sellavel** [comment]
@christianwen Tests well for me, can you check one more case. One of the changes proposed here causing a regression.

Steps: With code changes proposed above.

1. Log out 
2. Sign in again 
3. Open Search keyboard automatically closes on autofocus.

This happens until user manually focus the search input by tapping it.

**Comment by Santhosh-Sellavel** [comment]
Ah sorry, I missed to post it last week.

@christianwen proposal looks good to me, just waiting for [this](https://github.com/Expensify/App/issues/10731#issuecomment-1281182589)

Just one question to @iwiznia should fix in `react-native` & `react-native-web` library should go against our fork or library repoi?

**Comment by Santhosh-Sellavel** [comment]
Asked [here](https://expensify.slack.com/archives/C02NK2DQWUX/p1666027038603469)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by Santhosh-Sellavel** [comment]
> @christianwen Tests well for me, can you check one more case. One of the changes proposed here causing a regression.
> 
> Steps: With code changes proposed above.
> 
> 1. Log out
> 2. Sign in again
> 3. Open Search keyboard automatically closes on autofocus.
> 
> This happens until user manually focus the search input by tapping it.

@christianwen Did you check this?

**Comment by christianwen** [comment]
@Santhosh-Sellavel will do today

**Comment by christianwen** [comment]
@Santhosh-Sellavel I seem to see the cause of it but since this is a new code part, I'll need a bit more time to find the solution, will update tomorrow
thanks

**Comment by christianwen** [comment]
@Santhosh-Sellavel so the https://github.com/Expensify/App/issues/10731#issuecomment-1281182589 is due to the Composer autofocus behavior is conflicting with the Search input autofocus behavior, causing the Search input to lose focus automatically after gaining autofocus. This happens because the Composer is rendering even though the report screen is not active. So we'll prevent that by the below code in `src/pages/home/report/ReportActionCompose.js`
```diff
render() {
        // Waiting until ONYX variables are loaded before displaying the component
        if (_.isEmpty(this.props.personalDetails)) {
            return null;
        }

+        // only renders composer when the report screen is focused to avoid the composer autofocus behavior conflicting with other screen's autofocus behavior
+        if (!this.props.isFocused) {
+            return null;
+        }
```

**Comment by Santhosh-Sellavel** [comment]
Thanks, @christianwen, can you is there any other solution for this?  Not confident about this solution [here](https://github.com/Expensify/App/issues/10731#issuecomment-1285863019)

**Comment by Santhosh-Sellavel** [comment]
Can we off `autofocus` when it's not focused?

**Comment by Santhosh-Sellavel** [comment]
But this is also a bigger problem.
> This happens because the Composer is rendering even though the report screen is not active.

**Comment by Santhosh-Sellavel** [comment]
@iwiznia 
So far, 

We initiated a discussion [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1662989446760309) about how we want to handle and we end up with the following as-expected behavior
> On every time we open the chat always focus the input, but prevent the keyboard from opening on mobile & mWeb

I like @christianwen's proposal here https://github.com/Expensify/App/issues/10731#issuecomment-1255955313 & https://github.com/Expensify/App/issues/10731#issuecomment-1276991390

The proposal works well, while testing this I noticed something weird
> Open Search keyboard automatically closes on autofocus. (Breakage)

And the RC as mentioned [here](https://github.com/Expensify/App/issues/10731#issuecomment-1285863019) and there is a solution posted here

> @Santhosh-Sellavel so the https://github.com/Expensify/App/issues/10731#issuecomment-1281182589 is due to the Composer autofocus behavior conflicting with the Search input autofocus behavior, causing the Search input to lose focus automatically after gaining autofocus. This happens because the Composer is rendering even though the report screen is not active. So we'll prevent that by the below code in src/pages/home/report/ReportActionCompose.js
> ```diff
> render() {
>         // Waiting until ONYX variables are loaded before displaying the component
>         if (_.isEmpty(this.props.personalDetails)) {
>             return null;
>         }
> 
> +        // only renders composer when the report screen is focused to avoid the composer autofocus behavior conflicting with other screen's autofocus behavior
> +        if (!this.props.isFocused) {
> +            return null;
> +        }
> ```

What is your thought on the proposal and solution to the breakage?

**Comment by Santhosh-Sellavel** [comment]
> Is this still open for proposals?

Yes and our expected behavior is
`On every time we open the chat, Always focus the input but prevent the keyboard from opening on mobile & mWeb`


> 
> > Log out
> > Sign in again
> > Open Search keyboard automatically closes on autofocus.
> 
> @Santhosh-Sellavel question about this: on which platform does this happen? It would be good to share the video.

@0xmiroslav this happened on Android/iOS while using applying @christianwen's proposal.

**Comment by mallenexpensify** [comment]
@Santhosh-Sellavel can you address @christianwen 's comment above please? 
https://github.com/Expensify/App/issues/10731#issuecomment-1286691363

**Comment by Santhosh-Sellavel** [comment]
@mallenexpensify I'm waiting for @iwiznia thoughts on this https://github.com/Expensify/App/issues/10731#issuecomment-1286086036

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by iwiznia** [comment]
> I like @christianwen's proposal here #10731 (comment) & #10731 (comment)

Sounds good to me too.

> The proposal works well, while testing this I noticed something weird
> Open Search keyboard automatically closes on autofocus. (Breakage)

What do you mean by `open search keyboard`?

**Comment by Santhosh-Sellavel** [comment]
> > The proposal works well, while testing this I noticed something weird
> > Open Search keyboard automatically closes on autofocus. (Breakage)
> > 
> 
> What do you mean by open search keyboard?
@iwiznia 
While Opening Search from ChatList, The keyboard is dismissed automatically

**Comment by Santhosh-Sellavel** [comment]
> While Opening Search from ChatList, The keyboard is dismissed automatically

This is not reproducible, So I think let us move on, can you check this and confirm @christianwen? Maybe some recent changes solved this one.

**Comment by christianwen** [comment]
@Santhosh-Sellavel yes, I've checked and it's not reproducible any more

**Comment by mallenexpensify** [comment]
@Santhosh-Sellavel where are we at with this?  For the 'not reproducible' comment above, that's only for part of the bug/issue/proposal... not the whole issue, right?

**Comment by Santhosh-Sellavel** [comment]
Yeah Let's move forward. 
@iwiznia Good with @christianwen proposal!

cc: @mallenexpensify

**Comment by iwiznia** [comment]
Yep, @mallenexpensify can you hire @christianwen please?

**Comment by christianwen** [comment]
I've applied to the Upwork job

**Comment by Santhosh-Sellavel** [comment]
@christianwen PRs should go against to library's repo. Not against our fork, let us know when PR is raised, and keep us informed on progress thanks!

**Comment by christianwen** [comment]
thanks for sharing the video @mallenexpensify, it looks really awesome!

**Comment by christianwen** [comment]
Since this involves many repos, I'll get the PRs ready over the weekends. Thanks

**Comment by christianwen** [comment]
For `react-native-web` fix , the issue was raised here https://github.com/necolas/react-native-web/issues/2421 but it seems like the maintainer has different suggestion to resolve it, and a patch was created for `react-native` was created instead https://github.com/facebook/react-native/pull/35228. This will result in some small changes to my original proposal. I'll update accordingly after that RN patch is merged

**Comment by Santhosh-Sellavel** [comment]
Thanks for the update @christianwen

**Comment by christianwen** [comment]
PR for `react-native` patch created here https://github.com/facebook/react-native/pull/35244, awaiting review

**Comment by mallenexpensify** [comment]
@Christinadobrzyn Reassigning as I'm heading OOO for a month

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by Santhosh-Sellavel** [comment]
> PR for `react-native` patch of the `showSoftInputOnFocus` iOS behavior created here [facebook/react-native#35244](https://github.com/facebook/react-native/pull/35244), awaiting review

Seems like this got merged 🎉, Let's get this over the line, whats next here? @christianwen

**Comment by christianwen** [comment]
@Santhosh-Sellavel it was merged but wasn't released yet, we'll only pull 'react-native' to our fork once it's released right?

**Comment by Santhosh-Sellavel** [comment]
Can you ask it on slack, I'm not aware of that!

**Comment by christianwen** [comment]
asked here https://expensify.slack.com/archives/C01GTK53T8Q/p1668700495132809
@iwiznia are you aware of this?

**Comment by JmillsExpensify** [comment]
I think @christianwen is right, though thanks for taking this discussion to Slack to make sure we're on the same page.

**Comment by iwiznia** [comment]
Honestly, I am a bit lost on the react native fixes... I _think_ we don't need for a release to put it in our fork, do we? If we wait for a release then we could use the non forked version.

This is explained here https://stackoverflow.com/c/expensify/questions/12761/12762#12762 (which I think you don't have access to, so pasting here... damn, sorry but styling was not copied): 
```
What is the process for working with a fork?

Create the fork in the Expensify GitHub org. This requires GitHub admin permissions, which is only for @ring0 peeps. If that's not you, create a GitHub issue in Expensify/Expensify asking someone to fork the repo for you and tag it with @ring0.
Submit a PR including the new feature or bug fix against the main/master branch of the Expensify fork of the repo.
Note: Internal contributors can clone the forked repo, create a branch, and submit a PR against main like they would with any of our other repos. External contributors need to make their own fork of the upstream repo, then submit a PR from their fork to the Expensify fork.
Review and merge the PR in the Expensify fork.
Update our app to use the Expensify forked version of the dependency instead of the upstream.
Note: Sometimes this will be straightforward, but other times you may need to build the fork as part of our deploy process. This step might look very different for all kinds of dependencies. The happy path is to just point package.json at the latest commit of the main branch in the Expensify fork, but generally that will only work for JavaScript-only changes, not TypeScript or native code changes.
Submit a PR from the Expensify fork to the upstream.
Once the upstream PR is merged and released, point our app back at the upstream (which now includes the fix).
Lastly, merge changes from the upstream back into our fork, resolving any conflicts to use the changes from the upstream. Once this step is done, there should be no diff between the fork and the upstream.
If the maintainers of the upstream are very responsive or the bug is not urgent, we can choose to skip some steps:

Submit a PR from any fork to the upstream.
Wait for the upstream PR to be merged and released
Note: During this time we are at the mercy of people beyond our control – could be days, months, or never. We don't really know.
Once the upstream PR is merged, update the dependency in our app.
Dealing with forks is a bit of a pain, so I often opt for a middle-ground:

Submit a PR from any fork to the upstream.
Wait some time to give the maintainers and opportunity to review the PR.
This typically might be 2-3 weeks, but use your judgment based on how actively the repo is maintained, the quality of the bug report, and the severity of the bug.
If there's no activity in the upstream and you still think the fix is good, then fork the dependency in the Expensify org.
Submit a duplicate PR against the Expensify fork.
Merge the PR in our fork.
Point our app at the fork.
Wait for the upstream PR to be reviewed, merged and released.
Note: If the maintainer requests changes on the PR during this stage, those same changes should also be applied to our fork!
Then point NewDot back at the upstream.
Lastly, merge changes from the upstream back into our fork, resolving any conflicts to use the changes from the upstream. Once this step is done, there should be no diff between the fork and the upstream.
```

**Comment by Santhosh-Sellavel** [comment]
@christianwen Do you understand?

**Comment by christianwen** [comment]
@Santhosh-Sellavel yes
@iwiznia  
I see in the your pasted doc there's `Wait for the upstream PR to be merged **and released**`. So I think we might still need to wait for the release?
If not, should I be the one to create the PR to sync the upstream react-native to our forked react-native? I'm not sure on the process here since the upstream sync might involve a lot more changes than just the change relevant to this ticket

**Comment by iwiznia** [comment]
> This typically might be 2-3 weeks, but use your judgment based on how actively the repo is maintained, the quality of the bug report, and the severity of the bug.
> If there's no activity in the upstream and you still think the fix is good, then fork the dependency in the Expensify org.

Let's send the PR upstream and see how responsive they are?

**Comment by christianwen** [comment]
@iwiznia  we already sent the PR upstream and it was merged here https://github.com/facebook/react-native/pull/35244
The part that we’re not sure is whether to wait for the react-native official release to pull the changes back to our fork

**Comment by iwiznia** [comment]
ah, so it was merged 14 days ago, but it is still not on any released version?

**Comment by Santhosh-Sellavel** [comment]
Yeah, still not released!

**Comment by iwiznia** [comment]
ok, if next week we see no movement, we can use ou fork.

**Comment by Santhosh-Sellavel** [comment]
@christianwen Do we need to update the react-native-web?

**Comment by christianwen** [comment]
@Santhosh-Sellavel yes, 'react-native-web' needs to be updated by its maintainer to support the 'inputMode' property, which I asked here https://github.com/necolas/react-native-web/issues/2421#issuecomment-1319881454 but haven't got response yet, will follow up

**Comment by Santhosh-Sellavel** [comment]
@christianwen Any update?

**Comment by christianwen** [comment]
@Santhosh-Sellavel unfortunately I still got no response from the maintainer of react-nativr-web, last ping was https://github.com/necolas/react-native-web/issues/2421#issuecomment-1325345191. In this case can we merge the changes to our react-native-web fork instead?
cc @iwiznia

**Comment by iwiznia** [comment]
Yeah, I think that's best at this point, right? Seems the PR will be merged, just not right now, no?

**Comment by christianwen** [comment]
@iwiznia it's not really that PR content, but the 'inputMode' support mentioned in https://github.com/necolas/react-native-web/issues/2421#issuecomment-1308019532, it will be added to 'react-native-web' eventually, just that we haven't been able to confirm with the maintainer on the dates, so we might want to add that 'inputMode' to our own fork for now

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by iwiznia** [comment]
They said:
> Support for inputMode='none' is now merged into react-native@main. Not sure if it will make it into the react-native@0.71.0 release. But inputMode will be supported in react-native-web@0.19.

So if I get this right, this means:
- react-native has support for it, but not in a released version. I assume we can add those fixes to our fork?
- react-native-web does not have support for it yet, but they plan to add it for version 0.19. We would need to wait till this is properly implemented there, but we don't care much about that, since this issue is only for ios/android and not for web

Is this correct? And if not, what is exactly what you are proposing on doing?

**Comment by Santhosh-Sellavel** [comment]
> They said:
> 
> > Support for inputMode='none' is now merged into react-native@main. Not sure if it will make it into the react-native@0.71.0 release. But inputMode will be supported in react-native-web@0.19.
> 
> So if I get this right, this means:
> 
> * react-native has support for it, but not in a released version. I assume we can add those fixes to our fork?
> * react-native-web does not have support for it yet, but they plan to add it for version 0.19. We would need to wait till this is properly implemented there, but we don't care much about that, since this issue is only for ios/android and not for web
> 
> Is this correct? And if not, what is exactly what you are proposing on doing?

We need this for mWeb too, that's why react-native-web is necessary here!

**Comment by christianwen** [comment]
@Santhosh-Sellavel @iwiznia we've been waiting for the `react-native` release and `react-native-web` response from maintainer for some time now. Do you think it's ok to merge the fixes to our fork first then sync with the above libs later when they are updated?
Otherwise we'll need to continue waiting.

**Comment by iwiznia** [comment]
Let's wait, using the forks is a lot more work and this bug is not that bad anyway.

**Comment by Santhosh-Sellavel** [comment]
Still on hold, @christianwen follow this and let us know when its released!

**Comment by christianwen** [comment]
the `react-native` changes are still not released in upstream
same for `react-native-web`, still no response from the maintainer regarding expected release date https://github.com/necolas/react-native-web/issues/2421#issuecomment-1348470641

**Comment by iwiznia** [comment]
Still waiting

**Comment by iwiznia** [comment]
Still waiting

**Comment by iwiznia** [comment]
Still nothing it seems....

**Comment by mallenexpensify** [comment]
Might it make sense to offer to compensate the maintainer of RNW to help prioritize a fix?

**Comment by Santhosh-Sellavel** [comment]
Still on hold!

**Comment by iwiznia** [comment]
Still on hold!

**Comment by Santhosh-Sellavel** [comment]
Still on hold

**Comment by Santhosh-Sellavel** [comment]
we can get this merged to our fork

**Comment by iwiznia** [comment]
Maybe that's best? It's been a while

**Comment by Santhosh-Sellavel** [comment]
@christianwen this is released now can we get back on track?

**Comment by christianwen** [comment]
@Santhosh-Sellavel if not mistaken we need to wait until [this PR](https://github.com/Expensify/App/pull/15130) to update our app RN version to 0.71+ to be merged.

The new field from the RNW is only available on 0.71+ as confirmed by the maintainer [here](https://github.com/necolas/react-native-web/issues/2421#issuecomment-1485771574).

**Comment by MelvinBot** [comment]
📣 @christianwen! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by Santhosh-Sellavel** [comment]
@iwiznia or @Christinadobrzyn Removed Hold please, thanks!

**Comment by iwiznia** [comment]
Done

**Comment by christianwen** [comment]
noted @Santhosh-Sellavel 
I'll catch up and post an update in 1 day.
Thanks!

**Comment by trjExpensify** [comment]
👋 Came across this in the WAQ +4 weeks old dashboard. Now that it's off hold and can be worked on, dropping it back to `daily`.

**Comment by Santhosh-Sellavel** [comment]
Sounds good 👍

**Comment by Santhosh-Sellavel** [comment]
@christianwen Any update on the next steps

**Comment by Santhosh-Sellavel** [comment]
@iwiznia Can we merge it to our fork?

@christianwen what about the react-native web?

**Comment by Santhosh-Sellavel** [comment]
@christianwen We can merge it our fork and move forward with this!

**Comment by christianwen** [comment]
@Santhosh-Sellavel PR is up for cherry picking the RN change to our fork: https://github.com/Expensify/react-native/pull/52

**Comment by christianwen** [comment]
> @christianwen what about the react-native web?

@Santhosh-Sellavel we need to update our App to use the `0.19.0` update of `react-native-web` https://github.com/necolas/react-native-web/releases/tag/0.19.0 as per [this comment](https://github.com/necolas/react-native-web/issues/2421#issuecomment-1485771574). I think we should do this in a separate PR from this issue since there might be other changes required, what do you think?

**Comment by iwiznia** [comment]
Agree on doing the update in a separate PR

**Comment by Santhosh-Sellavel** [comment]
We could cherry-pick the same on RNW too, can you spin a PR for the same? @christianwen

**Comment by christianwen** [comment]
@Santhosh-Sellavel We already have a PR to upgrade our RNW fork to `0.19` here https://github.com/Expensify/App/issues/16660 but it's on HOLD, should we wait for it?

**Comment by MelvinBot** [comment]
📣 @christianwen! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/231263719-a8a3b7d3-36a0-4d7f-8ad4-cebe1f560fa4.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by Santhosh-Sellavel** [comment]
> @Santhosh-Sellavel We already have a PR to upgrade our RNW fork to `0.19` here https://github.com/Expensify/App/issues/16660 but it's on HOLD, should we wait for it?

No we don't need I guess, there is no dependency for us there.

**Comment by christianwen** [comment]
Contributor details
Your Expensify account email: christianwen18+100@gmail.com
Upwork Profile Link: https://upwork.com/freelancers/~01991fd5e5c11ef3ba

**Comment by christianwen** [comment]
@Santhosh-Sellavel this is the commit to add the `inputMode` in RNW https://github.com/necolas/react-native-web/commit/42bf402929b02f0c254c6fc53508ae374531e79f, it also includes a number of other changes (`enterKeyHint`, `readOnly`, `rows`). Are we ok to cherry-pick that whole commit? 

If modifying it for only the `inputMode` change then it will cause conflicts later.

**Comment by Santhosh-Sellavel** [comment]
I think we should wait for migration here. Cc: @iwiznia your thoughts?

**Comment by iwiznia** [comment]
I think I agree.

**Comment by Santhosh-Sellavel** [comment]
Thanks, @iwiznia. Let's add hold again when both changes are available will just resume and get done with it.

**Comment by MelvinBot** [comment]
@iwiznia, @christianwen, @Christinadobrzyn, @Santhosh-Sellavel Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by Santhosh-Sellavel** [comment]
This should be on hold @iwiznia @Christinadobrzyn

**Comment by Christinadobrzyn** [comment]
Going to move this issue to weekly while we are on hold - feel free to move it back to daily if that's better!

**Comment by iwiznia** [comment]
Still held

**Comment by Santhosh-Sellavel** [comment]
on hold

**Comment by Christinadobrzyn** [comment]
@Santhosh-Sellavel could you confirm which issue this is on hold for?

**Comment by Santhosh-Sellavel** [comment]
For next React native & React native web upgrade.

**Comment by Christinadobrzyn** [comment]
Still on hold for next React native & React native web upgrade.

**Comment by Christinadobrzyn** [comment]
We're on hold for this https://github.com/Expensify/App/issues/16660 right @Santhosh-Sellavel?

**Comment by Santhosh-Sellavel** [comment]
Yes

**Comment by Christinadobrzyn** [comment]
On hold for https://github.com/Expensify/App/issues/16660

**Comment by Christinadobrzyn** [comment]
On hold for https://github.com/Expensify/App/issues/16660

**Comment by iwiznia** [comment]
On hold for https://github.com/Expensify/App/issues/16660

**Comment by Christinadobrzyn** [comment]
I'm going to be ooo until August 31st so going to unassign and assign a new teammate. 

At this time, we're on hold for https://github.com/Expensify/App/issues/16660

**Comment by iwiznia** [comment]
Going on sabbatical, please reassign another engineer if this goes off hold

**Comment by Christinadobrzyn** [comment]
back from ooo - Thanks for handling this @NicMendonca! I can take it off your plate. 

Looks like we're still on hold for https://github.com/Expensify/Expensify/issues/302294

**Comment by Christinadobrzyn** [comment]
This appears to be fixed! I'll ask QA to test just to make sure -

**Comment by Santhosh-Sellavel** [comment]
@christianwen Can you confirm whether our upstream PRs got released and were it included in the recent react native migration and what the next step is here?

**Comment by Christinadobrzyn** [comment]
Hey @Santhosh-Sellavel would you be able to see if this GH will be fixed with this job? It doesn't seem to be exactly the same but it seems like the root might be the same? https://github.com/Expensify/App/issues/24314

**Comment by Santhosh-Sellavel** [comment]
That's a bit different from this issue as it happens in the other flow. Please check whether it's reproducible before proceeding?

**Comment by Santhosh-Sellavel** [comment]
> @christianwen Can you confirm whether our upstream PRs got released and were it included in the recent react native migration and what the next step is here?

bump @christianwen

**Comment by Santhosh-Sellavel** [comment]
> @Santhosh-Sellavel [This issue](https://github.com/Expensify/App/issues/16660) that we depend on is off HOLD now but not done yet, we need to wait until it’s done (means react-native-web is updated to v0.19 or greater)

@christianwen For mWeb alone we need to wait or what? 

As it's been held for more than 6 months, can you just summarize whats the status & what will be the next steps here when it comes off hold? So we are clear about the action plan here, thanks!

**Comment by Christinadobrzyn** [comment]
not overdue - @christianwen can youprovide a summary based on @Santhosh-Sellavel request here - https://github.com/Expensify/App/issues/10731#issuecomment-1675135307

**Comment by Christinadobrzyn** [comment]
I think we're waiting on this - https://github.com/Expensify/App/issues/16660

**Comment by Christinadobrzyn** [comment]
Going to move this to weekly while we wait for https://github.com/Expensify/App/issues/16660

**Comment by Santhosh-Sellavel** [comment]
On Hold

**Comment by Christinadobrzyn** [comment]
Monitoring https://github.com/Expensify/App/issues/16660

**Comment by Christinadobrzyn** [comment]
Monitoring https://github.com/Expensify/App/issues/16660

**Comment by Christinadobrzyn** [comment]
Monitoring https://github.com/Expensify/App/issues/16660

**Comment by iwiznia** [comment]
Still holding on $16660

**Comment by iwiznia** [comment]
Still holding on #16660

**Comment by iwiznia** [comment]
This is off hold since #16660 was already merged

**Comment by Santhosh-Sellavel** [comment]
@christianwen Can we get an update here?

**Comment by Santhosh-Sellavel** [comment]
bump @christianwen any update?

**Comment by Santhosh-Sellavel** [comment]
@christianwen can we get an update this week? Let us know thanks!

**Comment by Christinadobrzyn** [comment]
cc @Santhosh-Sellavel ^ https://github.com/Expensify/App/issues/10731#issuecomment-1849571489

**Comment by Christinadobrzyn** [comment]
hey @Santhosh-Sellavel how is the PR review going? https://github.com/Expensify/App/pull/32711

**Comment by Santhosh-Sellavel** [comment]
@Christinadobrzyn 
Sorry for the delay!

🤦 Kind of caught up in a loop of conflicts or missing this out. Will continue to review once conflicts are resolved.

Pinned this to the top of my list, this is the oldest issue it has been on hold for months.

**Comment by Santhosh-Sellavel** [comment]
Waiting for Review changes.

I'm unavailable early next week, please hold on to this for me, as I've been assigned to this for a long time.

cc: @Christinadobrzyn

**Comment by Christinadobrzyn** [comment]
Sounds good @Santhosh-Sellavel I added a [hold] label to the title. Let us know when you're back!

**Comment by Christinadobrzyn** [comment]
hey @Santhosh-Sellavel just checking in on this! Let me know if you have an update!

**Comment by Santhosh-Sellavel** [comment]
I'm Available now, waiting for update on PR

**Comment by Christinadobrzyn** [comment]
PR in the works - https://github.com/Expensify/App/pull/32711

**Comment by Santhosh-Sellavel** [comment]
Update:
@Christinadobrzyn 

This issue open for more than one and half years. And has been on hold for most of the time.
When we made a hiring decision, the following was our expected outcome for the composer

> Always focus the input, but prevent keyboard from opening on mobile & mWeb

To achieve we had two upstream PR's on react-native & react-native-web. It was mostly on hold waiting for the upstream release.

In the meanwhile some improvements seems to be made for the same composer on focus aspect. What we might do here now might not be the new expected outcome. So asked for a comparision from @christianwen i.e between current and new behaviour post changes on the PR. We will get consensus to decide are we good to go  & move forward from there, thanks!

cc: @iwiznia

**Comment by trjExpensify** [comment]
What's the update here? Do we still need a hold in the title?

**Comment by Christinadobrzyn** [comment]
Thanks for the nudge @trjExpensify - I should have removed the hold around this time because there's a PR happening, so I'll do that now - sorry for missing that. 

@Santhosh-Sellavel can you tell us the plan with this one based on the details here - https://github.com/Expensify/App/issues/10731#issuecomment-1997924006

**Comment by Christinadobrzyn** [comment]
Hi @Santhosh-Sellavel can you provide an update? I know this has been on hold for a few times, do we have a PR we're currently waiting on for this?

**Comment by Santhosh-Sellavel** [comment]
@Christinadobrzyn 

Mostly this PR was on hold for upstream releases to address this in both native & mWeb platforms. After release we started working & testing this.

But we run into a new blocker that breaks the expected behavior on the mWeb platforms due to changes made on App while we were on hold & Reviewing.

We need to create a new issue for that might need to put this on hold until its resolved.

Or Try to merge the changes if it solves for native need to decide on that.

**Comment by Christinadobrzyn** [comment]
Ah, thank you, @Santhosh-Sellavel. Has the blocker been reported in the Expensify-Bugs channel? It might be best to report the blocker to QA so they can create a GH that can be worked on. Then we can place a hold on this while that GH is worked on. 

Does that sound like the best next steps? 

Or, in terms of the merge option, is that something we should discuss in Slack with the team?

**Comment by Santhosh-Sellavel** [comment]
> Ah, thank you, @Santhosh-Sellavel. Has the blocker been reported in the Expensify-Bugs channel? It might be best to report the blocker to QA so they can create a GH that can be worked on. Then we can place a hold on this while that GH is worked on.
> Does that sound like the best next steps?

Yeah pretty much but not sure whether we need to log the bug in the https://github.com/Expensify/react-native-live-markdown or this project

> Or, in terms of the merge option, is that something we should discuss in Slack with the team?|

Yes waiting for @christianwen to fix non-mWeb issues.

**Comment by iwiznia** [comment]
Oh, was waiting for @Santhosh-Sellavel and @Christinadobrzyn to chime in here, my comment is already [here](https://github.com/Expensify/App/pull/32711#issuecomment-2102671859)

**Comment by iwiznia** [comment]
> Put on hold wait for other issues to be resolves

I am fine with this too, but not sure what exactly we would be holding on, can you spell that out for me?

**Comment by iwiznia** [comment]
Who is the new C+ and where do you want me to assign them?

**Comment by iwiznia** [comment]
Ah he's ooo now? Yes, we probably should, but not sure how to retrigger the assignment... @Christinadobrzyn do you know?

**Comment by trjExpensify** [comment]
Just ask in #contributor-plus for someone available.

**Comment by DylanDylann** [comment]
@trjExpensify I am interested in taking this issue as C+. Will try to speed up this one because it delayed for a long time. Thanks

**Comment by trjExpensify** [comment]
Sorry, been out for a few days. Done!

**Comment by Christinadobrzyn** [comment]
Ah thanks @trjExpensify! @DylanDylann @christianwen let us know if there's any update on this!

**Comment by DylanDylann** [comment]
We are reviewing the PR

**Comment by Christinadobrzyn** [comment]
just a heads up that I'm going to be ooo 9/4 - 9/10. I don't think we need another BZ teammate on this so just going to keep myself assigned. If you do, please reach out to the team and they can add someone.

**Comment by Christinadobrzyn** [comment]
Looks like PR is still in review - https://github.com/Expensify/App/pull/32711

@DylanDylann let us know if there's any specific update

**Comment by DylanDylann** [comment]
@christianwen I will give this high priority and will respond as soon as possible. Let's be active in the next few days to get it done. It has been around for over 2 years

**Comment by DylanDylann** [comment]
@christianwen Any update on the PR?

**Comment by Christinadobrzyn** [comment]
Update for Melvin - PR is in the works - https://github.com/Expensify/react-native-live-markdown/pull/499

**Comment by Skalakid** [comment]
Hello, we've @christianwen's [PR](https://github.com/Expensify/react-native-live-markdown/pull/499) into the `react-native-live-markdown` library (version `0.1.156`). I have to bump the version of the library to `0.1.157` in my [PR](https://github.com/Expensify/App/pull/49250), so I will add the test steps of this issue to the description. Is that okay for you?

**Comment by Christinadobrzyn** [comment]
just a heads up that I'm going to be ooo until 9/30. I'm not going to assign a BZ teammate since we're working on the PR/discussing the issue. If you need someone before Monday please reach out in Slack for a volutneer

**Comment by Christinadobrzyn** [comment]
Just catching up here - based on this [convo](https://github.com/Expensify/App/issues/10731#issuecomment-2363751533). We are waiting on https://github.com/Expensify/App/pull/49250, is that correct @Skalakid @christianwen

**Comment by iwiznia** [comment]
Reverted the PR: https://github.com/Expensify/App/pull/32711#issuecomment-2386339013

**Comment by DylanDylann** [comment]
> Hello, we've @christianwen's https://github.com/Expensify/react-native-live-markdown/pull/499 into the react-native-live-markdown (version 0.1.156). I have to bump the version of the library inside E/App to 0.1.157 in my https://github.com/Expensify/App/pull/49250, so @christianwen can create his PR without the need to bump the library but it will need to be on hold after we merge https://github.com/Expensify/App/pull/49250

@christianwen Please bump the version of react-native-live-markdown lib to 0.1.156 in our PR because @Skalakid's PR has been reverted. I think we should handle it separately to avoid blocking our PR

**Comment by Skalakid** [comment]
@DylanDylann, I'm not sure if it's a good idea. In version 0.1.146, we've added the inline images feature, and we need to bump the library together with all the library fixes added after this version + with some Expensify App changes. [Here](https://github.com/Expensify/App/pull/50047) I created another PR with all relevant fixes. Please wait until it's merged

**Comment by Christinadobrzyn** [comment]
This is a super old GH so please let me know if I'm missing anything (as I was just added)

Prepping for payment so I don't lose track of the regression.

Payouts due:

- [ ] Contributor: $1000 (50% reduction for regression) @christianwen 
- [ ] Reviewer: $1000 (50% reduction for regression) @DylanDylann

**Comment by DylanDylann** [comment]
@Christinadobrzyn The PR still be reviewing

**Comment by Christinadobrzyn** [comment]
Gonna move this back to weekly while we wait for the regression PR to go into production and have a new payment date.

**Comment by Christinadobrzyn** [comment]
Note for Melvin - [PR](https://github.com/Expensify/App/pull/51172) under review cc @DylanDylann

**Comment by Christinadobrzyn** [comment]
Just checking on this @DylanDylann can you provide an update on the [PR](https://github.com/Expensify/App/pull/51172) review? Thanks!

**Comment by DylanDylann** [comment]
I am reviewing the PR

**Comment by Christinadobrzyn** [comment]
monitoring PR - https://github.com/Expensify/App/pull/51172

**Comment by Christinadobrzyn** [comment]
monitoring PR - https://github.com/Expensify/App/pull/51172

**Comment by Christinadobrzyn** [comment]
monitoring PR - https://github.com/Expensify/App/pull/51172

**Comment by Christinadobrzyn** [comment]
monitoring PR - https://github.com/Expensify/App/pull/51172

**Comment by Christinadobrzyn** [comment]
monitoring PR - https://github.com/Expensify/App/pull/51172

**Comment by Christinadobrzyn** [comment]
@DylanDylann in preparation for payment day - let us know if we need a regression test for this!

**Comment by DylanDylann** [comment]
# BugZero Checklist:

- [ ] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1.

Do we agree 👍 or 👎


</details>

**Comment by DylanDylann** [comment]
I will post a checklist when the regression period is gone. In this issue, We still have some problems that need to be solved

**Comment by DylanDylann** [comment]
# BugZero Checklist:

- [ ] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1.

Do we agree 👍 or 👎


</details>

**Comment by Christinadobrzyn** [comment]
I updated the regression test a little bit - @DylanDylann can you review these steps to make sure they are accurate for both devices?

### iOS testing

- Open the iOS mobile app
- Go to any chat
- Verify that the composer is focused and the virtual keyboard is not shown.
- After touching on the composer, the keyboard is shown

### Android testing

- Open the iOS mobile app
- Go to any chat
- Verify that the composer is focused and the virtual keyboard is not shown.
- After touching on the composer, the keyboard is shown

**Comment by Christinadobrzyn** [comment]
Is this really 2 years old?

Payment summary here - https://github.com/Expensify/App/issues/10731#issuecomment-2592911084, can you please check to make sure this is correct since this is so old?

@christianwen and @DylanDylann are you both paid in Upwork? @christianwen can you send me your upwork profile?

**Comment by DylanDylann** [comment]
> Is this really 2 years old?

With approximately 550 comments on this issue 😄

**Comment by DylanDylann** [comment]
@Christinadobrzyn The regression tests are fine

**Comment by Christinadobrzyn** [comment]
Awesome, thanks @christianwen and @DylanDylann! 

@DylanDylann can you please accept this offer? https://www.upwork.com/freelancers/~010c6310ab55f3d17d

**Comment by DylanDylann** [comment]
@Christinadobrzyn accepted, thanks 🙏

**Comment by Christinadobrzyn** [comment]
Awesome, thanks @DylanDylann! This is paid out in Upwork based on this payment summary - https://github.com/Expensify/App/issues/10731#issuecomment-2606658596

Closing as complete.
